pkg/services/datasources/service/store.go

```diff
@@ -285,6 +285,8 @@ func (ss *SqlStore) UpdateDataSource(ctx context.Context, cmd *datasources.Updat
 		sess.UseBool("basic_auth")
 		sess.UseBool("with_credentials")
 		sess.UseBool("read_only")
+		// Make sure database field is zeroed out if empty. We want to migrate away from this field.
+		sess.MustCols("database")
```

Comment:
Skeptic you want to do this before merging the other PR. Doesn't that risk clearing database column without writing database value to json data?

Reply:
I can't really see this as a risk but we can also do this in the other pr. Just wanted to separate them a little.

Reply:
Merging this PR would trigger a nightly build (at least used to do) and/or grafana cloud build so there's a chance this creates an actual problem if we're unlucky. Would suggest doing it in the other PR or merge them together

Reply:
Sorry for slow response btw. Just realized you've removed this now so will approve